Makefile

```diff
@@ -99,7 +99,7 @@ k0s.exe k0s: $(GO_SRCS)
 	touch $@
 
 .PHONY: lint
-lint: pkg/assets/zz_generated_offsets_$(TARGET_OS).go
+lint: pkg/assets/zz_generated_offsets_$(TARGET_OS).go lint-gomod
```

Comment:
Instead of the `GOMODTIDYLINT` variable, couldn't you just tie in the `SKIP_GOMOD_LINT` check here as a conditional dependency?

ie. `lint: pkg/assets/zz_generated_offsets_$(TARGET_OS).go $(if $(SKIP_GOMOD_LINT), , lint-gomod)`

This way, you can just leave the body of `lint-gomod` as unconditional.

Reply:
Done not exactly how you suggested, but very close to:
- only one definition for lint-gomod target
- list of deps for lint target is updated by ifneq check for the SKIP_GOMOD_LINT variable